webdav/pkg/service/v0/service.go

```diff
@@ -54,57 +72,213 @@ func (g Webdav) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // Thumbnail implements the Service interface.
 func (g Webdav) Thumbnail(w http.ResponseWriter, r *http.Request) {
-	tr, err := thumbnail.NewRequest(r)
+	tr, err := requests.ParseThumbnailRequest(r)
+	if err != nil {
+		g.log.Error().Err(err).Msg("could not create Request")
+		renderError(w, r, errBadRequest(err.Error()))
+		return
+	}
+
+	c := thumbnails.NewThumbnailService("com.owncloud.api.thumbnails", grpc.DefaultClient)
```

Comment:
Should be moved to service initialization in my opinion. @refs what do you think about that?

Reply:
Agree. Clients should be shared since they already have a connection pool, creating a client per request could end up in a file descriptor leak if I'm not mistaken